### PR TITLE
[AUTOMATIC] Release/5.1.0

### PR DESCRIPTION
**Pull request by RCGitBot**
**This is an automatic release.**

### New Features
* Paywalls with custom purchase and restore logic handlers (#3973) via James Borthwick (@jamesrb1)
### Bugfixes
* Prevent paywall PurchaseHandler from being cleared on rerender (#4035) via Josh Holtz (@joshdholtz)
* Update Purchase Tester for 5.0.0 (#4015) via Will Taylor (@fire-at-will)
### Dependency Updates
* Bump fastlane from 2.221.0 to 2.221.1 (#3977) via dependabot[bot] (@dependabot[bot])
### Other Changes
* Bring official `xcodes` back to CI (#4029) via Cesar de la Vega (@vegaro)
* Paywalls tester with sandbox purchases (#4024) via James Borthwick (@jamesrb1)
* Update v5 migration guide to contain current latest version (#4019) via Toni Rico (@tonidero)
* CI Build Docs Improvements (#4014) via Will Taylor (@fire-at-will)
* Use available resource class for backend-integration-tests-offline-job (#4013) via Will Taylor (@fire-at-will)
* Add `X-Preferred-Locales` header (#4008) via Cesar de la Vega (@vegaro)

### CODE REVIEW

File: Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift [hunk 1]
```diff
@@ -243,19 +243,16 @@ class OfflineStoreKit1IntegrationTests: BaseOfflineStoreKitIntegrationTests {
 
     @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
     func testCallToGetCustomerInfoWithPendingTransactionsPostsReceiptOnlyOnce() async throws {
-        // forceRenewalOfSubscription doesn't work well, so we use this instead
-        setShortestTestSessionTimeRate(self.testSession)
```
Comment (tonidero) [inline Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift:247]: I reverted this change before 5.0 but was brought back when we merged 5.0. This was causing tests to fail in my machine and CI

**Comment by fire-at-will** [comment]
Shouldn't this be a `5.0.1` release instead of `5.1.0`?

**Comment by joshdholtz** [comment]
> Shouldn't this be a `5.0.1` release instead of `5.1.0`?

This has support for observer mode (errr, naming) paywalls which is a whole new API so that is why this is being bumped by a minor version

**Comment by tonidero** [comment]
> Shouldn't this be a `5.0.1` release instead of `5.1.0`?

Well if we consider https://github.com/RevenueCat/purchases-ios/pull/3973 as a new feature it should be a minor, which I think it would be. Happy to chat more if you feel otherwise though!

**Review by joshdholtz — APPROVED** [review]
Pair reviewed with @fire-at-will 💪

**Pull request merged**